Backpack/BarChart/Classes/BarChartBar.swift

```diff
@@ -0,0 +1,181 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018-2020 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import UIKit
+
+@objcMembers
+@objc(BPKBarChartBar)
+public final class BPKBarChartBar: UIControl {
+
+    let backgroundView: UIView = UIView()
+    let barView: UIView = UIView()
+    let miniFlareView: BPKMiniFlareView = BPKMiniFlareView()
+    let titleLabel: BPKLabel = BPKLabel(fontStyle: .textBaseEmphasized)
+    let subtitleLabel: BPKLabel = BPKLabel(fontStyle: .textSm)
+    var barViewHeightAnchor: NSLayoutConstraint?
+    var miniFlareViewPositionAnchor: NSLayoutConstraint?
+    let backgroundViewColor = BPKColor.dynamicColor(withLightVariant: BPKColor.skyGrayTint06,
+                                                    darkVariant: BPKColor.blackTint03)
+    let noPriceColor = BPKColor.skyGrayTint03
+    let priceColor = BPKColor.primaryColor
+    let selectedColor = BPKColor.dynamicColor(withLightVariant: BPKColor.monteverde, darkVariant: BPKColor.glencoe)
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.setupViews()
+    }
+
+    public required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        self.setupViews()
+    }
+
+    func setupViews() {
+        self.backgroundView.backgroundColor = self.backgroundViewColor
```

Comment:
I don't think we should use `self` unless required to disambiguate. That's more swifty imo

Reply:
In addition to that, part of the code included here should live into the lazy getters for the properties.